src/fromager/sources.py

```diff
@@ -326,17 +336,30 @@ def _apply_patch(patch: pathlib.Path, source_root_dir: pathlib.Path):
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path
+    source_root_dir: pathlib.Path,
+    patches_dir: pathlib.Path,
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
+    # Get all the patches for that requirement
+    req_patches = overrides.patches_for_req(patches_dir, source_root_dir.name)
+
+    if len(req_patches) > 0:
+        # Set up regex to filter out unwanted patches.
+        # Example: instructlab and instructlab-sdg
+        pattern = re.compile(rf"^{req_name}(-v?\d+\.\d+\.\d+)?$")
```

Comment:
vllm will have 4 numerical parts to its version, and we have some packages that end with version `.post1`. If there are unversioned patches, in a directory that's the same as req_name, then those would have been applied and this function shouldn't be called, so I think you can ignore that case.

I think that's something like this:

```suggestion
        pattern = re.compile(rf"^{req_name}-v?(\d+\.)+\d+)")
```

It would be good to pull this regex out of the function and write some unit tests using known package names and versions that we have.

Reply:
Okay, I have added tests for regex and created a helper for it